src/app/core/services/label.service.ts

```diff
@@ -70,36 +94,91 @@ export class LabelService {
 
-    for (const label of labels) {
-      // Get the name and color of each label and store them into the service's array list
-      const labelName = String(label['name']).split('.');
-      const labelType = labelName[0];
-      const labelValue = labelName[1];
-      const labelColor = String(label['color']);
-
-      // Check for duplicate labels
-      if (this.labelColorMap.has(labelValue)) {
-        continue;
+    for (const category of Object.keys(this.REQUIRED_LABELS)) {
+      for (const labels of Object.values(this.REQUIRED_LABELS[category])) {
+        requiredLabels.push(labels as Label);
       }
+    }
 
-      this.labelColorMap.set(labelValue, labelColor);
-      const labelList = this.getLabelList(labelType);
+    return requiredLabels;
+  }
 
-      if (labelList !== undefined) {
-        labelList.push({labelValue: labelValue, labelColor: labelColor});
+  /**
+   * Ensures that the repo has the expected labels.
+   * Compares the actual labels in the repo with expected labels. If an expected label is missing,
+   * it is added to the repo. If the expected label exists but the label color is not as expected,
+   * the color is updated. Does not delete actual labels that do not match expected labels.
+   * i.e., the repo might have more labels than the expected labels after this operation.
+   * @param labels: JSON data representing labels in the repo.
+   * @param expectedLabels: expected labels.
+   */
+  private ensureRepoHasExpectedLabels(labels: Label[], expectedLabels: Label[]): void {
+
+    expectedLabels.forEach(label => {
+
+      // Finds for a label that has the same name as a required label.
+      const nameMatchedLabels: Label[] = labels.filter(remoteLabel =>
+          remoteLabel.getFormattedName() === label.getFormattedName());
+
+      if (nameMatchedLabels.length === 0) {
+        // Create new Label (Could not find a label with the same name & category)
+        this.githubService.createLabel(label.getFormattedName(), label.labelColor);
+      } else if (nameMatchedLabels.length === 1 && !nameMatchedLabels[0].equals(label)) {
+        // Update Label Color (Found a label with same name and category BUT different color.)
+        this.githubService.updateLabel(label.getFormattedName(), label.labelColor);
+      } else if (nameMatchedLabels.length > 1) {
+        throw new Error('Unexpected error: the repo has multiple labels with the same name ' + label.getFormattedName());
       }
 
-    }
-    // Sort the severity labels from Low to High
-    this.severityLabels.sort((a, b) => {
-      return SEVERITY_ORDER[a.labelValue] - SEVERITY_ORDER[b.labelValue];
+      this.saveLabelData(label);
     });
 
+    const sortingGroups: {labelArray: Label[], sortingFunction: (a: Label, b: Label) => number}[] = [];
+    sortingGroups.push({labelArray: this.labelArrays.severity,
+        sortingFunction: ((a, b) => SEVERITY_ORDER[a.labelValue] - SEVERITY_ORDER[b.labelValue])});
+    this.sortLabelArrays(sortingGroups);
```

Comment:
What's going on in this code block? It was not mentioned in the header comment.

Reply:
In the previous function prior to me editing it there was a section devoted to sorting the severity labels so that they appeared in the Low, Medium, High order in the app. Perhaps I could abstract this out to a separate method which contains sorting instructions for any list of labels.

Reply:
Removed the sort 👍